### PR TITLE
bpo-30534: Fix error messages when pass keyword arguments

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
to functions implemented in C that don't support this.

### CODE REVIEW

**Review by vstinner — APPROVED** [review]
LGTM, I also like included cleanup changes ;-)

On the issue, I requested unit tests. But I also proposed to change the error message, so we write tests later, when we agree on what should be done.

File: Lib/test/test_call.py [hunk 1]
```diff
@@ -33,9 +34,6 @@ def test_varargs2_ext(self):
         else:
             raise RuntimeError
 
-    def test_varargs0_kw(self):
-        self.assertRaises(TypeError, {}.__contains__, x=2)
-
```
Comment (vstinner) [inline Lib/test/test_call.py:38]: You may keep this test for other Python implementations. (This test doesn't check the error message.)

File: Lib/test/test_itertools.py [hunk 2]
```diff
@@ -1996,7 +1996,8 @@ def __init__(self, newarg=None, *args):
                 Subclass(newarg=1)
             except TypeError as err:
                 # we expect type errors because of wrong argument count
-                self.assertNotIn("does not take keyword arguments", err.args[0])
+                print(repr(err))
```
Comment (vstinner) [inline Lib/test/test_itertools.py]: Remove print()?

File: Objects/call.c [hunk 3]
```diff
@@ -637,16 +636,17 @@ _PyMethodDef_RawFastCallKeywords(PyMethodDef *method, PyObject *self, PyObject *
         break;
 
     case METH_VARARGS:
+        if (!(flags & METH_KEYWORDS) && nkwargs) {
```
Comment (vstinner) [inline Objects/call.c]: Is "!(flags & METH_KEYWORDS)" still needed?

File: Lib/test/test_call.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_call.py:38]: This is a duplicate of the following test.

File: Lib/test/test_itertools.py (hunk 2 above)
Comment (serhiy-storchaka) [reply Lib/test/test_itertools.py]: Done.

File: Objects/call.c (hunk 3 above)
Comment (serhiy-storchaka) [reply Objects/call.c]: Not needed. Removed.

File: Lib/test/test_call.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/test_call.py:38]: Oh ok, you are right.

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by serhiy-storchaka** [comment]
Thank you for your review @haypo.

**Pull request merged**